ui/src/main/resources/edu/wpi/grip/ui/templates/java/Pipeline.vm

```diff
@@ -0,0 +1,99 @@
+#**
+This creates the complete java pipeline.
+*##set($needsMut = false)
+#set($interpolation = false)
+#set($lines = false)
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.HashMap;
+
+import org.opencv.core.*;
+import org.opencv.core.Core.*;
+import org.opencv.features2d.FeatureDetector;
+import org.opencv.imgcodecs.Imgcodecs;
+import org.opencv.imgproc.*;
+
+/**
+* $fileName class.
+*
+* <P>Autogenerated pipeline from GRIP.
+*
+* <P>Make sure to set all sources using the setters before running process().
+*
+* @author GRIP
+*/
+public class $fileName{
+
+	//This map links the Outputs with their names
+	private Map<String,Object> outputs;
+#foreach($source in $pipeline.getSources())
+	private ${source.type()} $source.value();
+#end
```

Comment:
They are treated differently just to make the code more readable. A larger pipeline could easily have 10 to 15 outputs with only 1 or 2 inputs. From a user perspective it doesn't really matter how they are stored. If you think it would be cleaner to store them the same way, I can easily remove the map.